sbws/core/scanner.py

```diff
@@ -169,10 +170,11 @@ def measure_relay(args, conf, destinations, cb, rl, relay):
         cb.controller, conf.getfloat('general', 'http_timeout'))
     # Pick a destionation
     dest = destinations.next()
+    # If there is no any destination at this point, it can not continue.
     if not dest:
-        log.warning('Unable to get destination to measure %s %s',
-                    relay.nickname, relay.fingerprint[0:8])
-        return None
+        log.critical("There are not any functional destinations.")
+        # This should raise an error so that the caller can close the pool.
+        exit(1)
```

Comment:
What happens when all the pools fail?

Reply:
There's only one pool. The caller is the callback, i though it's better to just call  `stop_threads` (merged in master) from here (https://github.com/juga0/sbws/commit/3fe64b06253f9342e627c5d4b9db065cd03a8a7e, in https://github.com/juga0/sbws/tree/bug28897_squashed_rebased)